Src/Swagger/ValidationSchemaProcessor.cs

```diff
@@ -178,23 +178,27 @@ private void ApplyValidationRule(JsonSchema schema, IValidationRule validationRu
         {
             var propertyValidator = ruleComponent.Validator;
 
-            // 1. If the propertyValidator is a ChildValidatorAdaptor (RuleFor(....).SetValidator(new MyChildValidator())), we need to get the underlying validator
+            // 1. If the propertyValidator is a ChildValidatorAdaptor we need to get the underlying validator
+            // i.e. for RuleFor().SetValidator() or RuleForEach().SetValidator()
             if (propertyValidator.Name == "ChildValidatorAdaptor")
             {
                 // Get underlying validator using reflection
                 var validatorTypeObj = propertyValidator.GetType()
                     .GetProperty("ValidatorType")
                     ?.GetValue(propertyValidator);
                 // Check if something went wrong
-                if (validatorTypeObj is null || validatorTypeObj is not Type validatorType)
+                if (validatorTypeObj is not Type validatorType)
                     throw new InvalidOperationException("ChildValidatorAdaptor.ValidatorType is null");
 
                 // Retrieve or create an instance of the validator
                 if (!_childAdaptorValidators.TryGetValue(validatorType.FullName!, out var childValidator))
                     childValidator = _childAdaptorValidators[validatorType.FullName!] = (IValidator)Activator.CreateInstance(validatorType)!;
 
                 // Apply the validator to the schema. Again, recursively
-                ApplyValidator(schema.ActualProperties[propertyName].ActualSchema, childValidator, "");
+                var childSchema = schema.ActualProperties[propertyName].ActualSchema;
+                // Check if it is an array (RuleForEach()). In this case we need to apply validator to an Item Schema
+                childSchema = childSchema.Type == JsonObjectType.Array ? childSchema.Item.ActualSchema : childSchema;
+                ApplyValidator(childSchema, childValidator, string.Empty);
```

Comment:
in case of `RuleForEach()` the rules were applied but for a wrong schema... which didn't work obviously..

if you have `RuleForEach(x => x.Children).SetValidator(new ChildValidator())` the validator is for a single `Child` object.. but it was trying here to apply the rules for a `List<Child>` object